OSSM-5726 Distributed Tracing Docs Section Updates and Tempo content (2.5 Release) (#70517)
openshift/openshift-docs: modules/ossm-integrating-distr-tracing-tempo.adoc

////
This module is included in the following assemblies:
* service_mesh/v2x/ossm-observability.adoc
////

:_mod-docs-content-type: PROCEDURE
[id="ossm-integrating-distr-tracing-tempo_{context}"]
= Integrating distributed tracing platform (Tempo)
//this one is specific to Tempo
//another module may be specific to OTEL
//another module may be specific to Kiali
//they should all reside in ossm-observability, and in the order they need to happen/be completed

You can expose tracing data to the {DTProductName} (Tempo) stack by appending a named element and `zipkin` provider to the `spec.meshConfig.extensionProviders` specification. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.

The following example configures the `ServiceMeshControlPlane` resource to define an extensionProvider `zipkin` for sending tracing spans to a TempoStack. A TempoStack instance can be created in the `tracing-system` namespace after creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.

.Prerequisites

> You have created a TempoStack instance using the `Red Hat Tempo Operator` in the `tracing-system` namespace. For more information, see "Installing the distributed tracing platform (Tempo)". 

Reviewer: I think creating a Tempo Stack instance is not a Prerequisite. You can check the description above.
 
" A TempoStack instance can be created in the `tracing-system` namespace **after** creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources."

If a user creates a TempoStack instance before creating SMCP and SMMR, there will be a 401 access error issue.

Suggested change:
 ```suggestion
You have installed the `Red Hat Tempo Operator` and `Red Hat OpenShift Service Mesh Operator` in the `openshift-operators` namespace and have created namespaces such as `istio-system` and `tracing-system`.

```

.Procedure

> . Configure the `ServiceMeshControlPlane` resource to connect to the TempoStack instance in the `tracing-system` namespace:

Reviewer: ```suggestion
. Configure the `ServiceMeshControlPlane` resource to define an extension provider:

A TempoStack instance should be created **after** creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.
```

+
[source,yaml]
----
kind: ServiceMeshControlPlane
apiVersion: maistra.io/v2
metadata:
  name: basic
  namespace: istio-system
spec:
...
  meshConfig:
    extensionProviders:
      - name: tempo
        zipkin:
          service: tempo-sample-distributor.tracing-system.svc.cluster.local
          port: 9411
  tracing:
    sampling: 10000
    type: None <1>
  version: v2.5
---
kind: ServiceMeshMemberRoll
apiVersion: maistra.io/v1
metadata:
  name: default
  namespace: istio-system
spec:
  members:
    - tracing-system
----

> <1> The `spec.tracing.type` setting defines a Distributed Tracing instance and 2 Jaeger instances. Set `spec.tracing.type` to `None` when connecting to a TempoStack using an `extensionProvider` setting.

Reviewer: ```suggestion
<1> The `spec.tracing.type` setting defines a Distributed Tracing deprecated Jaeger instance. Set `spec.tracing.type` to `None` when connecting to a TempoStack using an `extensionProvider` setting.
```

////
This module is included in the following assemblies:
* service_mesh/v2x/ossm-observability.adoc
////

:_mod-docs-content-type: PROCEDURE
[id="ossm-integrating-distr-tracing-tempo_{context}"]

> = Integrating distributed tracing platform (Tempo)

Reviewer 2: I see TempoStack in our docs. Should it be...

```suggestion
= Integrating distributed tracing platform (TempoStack)
```

Consider this throughout

Author: The name of the product, according to the docs https://docs.openshift.com/container-platform/4.13/distr_tracing/distr_tracing_tempo/distr-tracing-tempo-installing.html its "Installing the distributed tracing platform (Tempo)"

Author: It's TempoStack when using it to reference "TempoStack instance," "TempoStack" resource, or "TempoStack" in the UI.

//this one is specific to Tempo
//another module may be specific to OTEL
//another module may be specific to Kiali
//they should all reside in ossm-observability, and in the order they need to happen/be completed

You can expose tracing data to the {DTProductName} (Tempo) stack by appending a named element and `zipkin` provider to the `spec.meshConfig.extensionProviders` specification. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.

.Prerequisites

> Install the {DTProductName} (Tempo) stack by following the instructions provided in 

Reviewer 2: I always suggest past tense for prerequisites: 

```suggestion
* You have installed the {DTProductName} (Tempo) stack by following the instructions provided in 
```

ifndef::openshift-rosa,openshift-dedicated[]

> xref:../../distr_tracing/distr_tracing_tempo/distr-tracing-tempo-installing.adoc[Installing the distributed tracing platform (Tempo)].

Reviewer 2: Can't have an xref in a module. This would need to be in quotes and then an additional resources section added to the module that included the xref. 

```suggestion
"Installing the distributed tracing platform (Tempo). "
```

endif::[]

.Procedure

> . Configure the `ServiceMeshControlPlane` resource to connect to the Tempo Stack instance in the `tracing-system` namespace:

Reviewer 2: afaic TempoStack should be one word. At least that's how it is in our docs: 

```suggestion
. Configure the `ServiceMeshControlPlane` resource to connect to the TempoStack instance in the `tracing-system` namespace:
```

> +

Reviewer 2: Is there a command to access this since it is in the `tracing-system` NS? Usually when altering a YAML file I say that there should be an accompanying command.

[source,yaml]
----
kind: ServiceMeshControlPlane
apiVersion: maistra.io/v2
metadata:
  name: basic
  namespace: istio-system
spec:
...
  meshConfig:
    extensionProviders:
      - name: tempo
        zipkin:
          service: tempo-sample-distributor.tracing-system.svc.cluster.local
          port: 9411
  tracing:
    sampling: 10000
    type: None <1>
  version: v2.5
---
kind: ServiceMeshMemberRoll
apiVersion: maistra.io/v1
metadata:
  name: default
  namespace: istio-system
spec:
  members:
    - tracing-system
----
<1> The `spec.tracing.type` setting defines a Distributed Tracing instance and 2 Jaeger instances. Set `spec.tracing.type` to `None` when connecting to a Tempo Stack using an `extensionProvider` setting.

> . Configure the Kiali resource specification to enable a Kiali workload Traces dashboard. You can use the dashboard to view tracing query results.

Reviewer 2: Same thought here. Is there an accompanying command?

> . Configure the Kiali resource specification to enable a Kiali workload Traces dashboard. You can use the dashboard to view tracing query results.

Reviewer 2: If this is an actual dashboard on the UI it would be need to be in asterisks. *Traces*. You might also tell users how to navigate here?

Author: Nope. Not done through a dashboard at this time.

+
[source,yaml]
----
apiVersion: kiali.io/v1alpha1
kind: Kiali
...
spec:  
  external_services:
    tracing:
      enabled: true
      in_cluster_url: 'http://tempo-sample-query-frontend.tracing-system.svc.cluster.local:16686'
      url: '[Tempo query frontend Route url]'
----
+

> . Apply a Telemetry custom resource for {SMProductShortName} to start the Tempo provider setting.

Reviewer 2: I believe that this would need an accompanying command to apply the CR. something like `oc apply -f` or something. 

Check the TempoStack vs. Tempo usage.

+
[source,yaml]
----
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: mesh-default
  namespace: istio-system
spec:
  tracing:
  - providers:
    - name: tempo
    randomSamplingPercentage: 100
----

> You can also create Istio Gateway and Virtual Service resources to expose an OpenShift Route for accessing the Tempo Jaeger Query console.

Reviewer 2: ```suggestion
You can also create Istio gateway and virtual service resources to expose an OpenShift route for accessing the Tempo Jaeger Query console.
```

////
This module is included in the following assemblies:
* service_mesh/v2x/ossm-observability.adoc
////

:_mod-docs-content-type: PROCEDURE
[id="ossm-integrating-distr-tracing-tempo_{context}"]
= Integrating distributed tracing platform (Tempo)
//this one is specific to Tempo
//another module may be specific to OTEL
//another module may be specific to Kiali
//they should all reside in ossm-observability, and in the order they need to happen/be completed

You can expose tracing data to the {DTProductName} (Tempo) stack by appending a named element and `zipkin` provider to the `spec.meshConfig.extensionProviders` specification. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.

> The following example configures the `ServiceMeshControlPlane` resource to define an extensionProvider `zipkin` for sending tracing spans to a TempoStack. A TempoStack instance can be created in the `tracing-system` namespace after creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.

Reviewer: resolve this. Some CI jobs overwrite bold format.

> The following example configures the `ServiceMeshControlPlane` resource to define an extensionProvider `zipkin` for sending tracing spans to a TempoStack. A TempoStack instance can be created in the `tracing-system` namespace after creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.

Author: Red Hat style is to use Italics instead of bold. Updated to comply with Red Hat style guide.
